app/grandchallenge/discussion_forums/models.py

```diff
@@ -45,7 +47,7 @@ class Topic(UUIDModel):
     )
 
     subject = models.CharField(max_length=255)
-    slug = AutoSlugField(populate_from="subject", max_length=255)
+    slug = AutoSlugField(populate_from="subject", max_length=64)
 
     TopicTypeChoices = TopicTypeChoices
     type = models.CharField(
```

Comment:
We shouldn't use `type` as a property as it clashes with the built-in. I think we use `kind` elsewhere to avoid this. Could you also add the database constraint on the choices?

With the boolean `locked` I think that using `is_locked` makes the code easier to read.